src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -224,6 +211,72 @@ impl LeaderBlockCommitOp {
             return Err(op_error::ParseError);
         }
 
+        let mut commit_outs = vec![];
+        let mut pox_fee = None;
+        let mut burn_fee = None;
+
+        for (ix, output) in outputs.into_iter().enumerate() {
+            // only look at the first OUTPUTS_PER_COMMIT outputs
+            //   or until first _burn_ output
+            if ix >= OUTPUTS_PER_COMMIT {
+                break;
+            }
+            if output.address.is_burn() {
+                burn_fee.replace(output.amount);
+                break;
+            } else {
+                // all pox outputs must have the same fee
+                if let Some(pox_fee) = pox_fee {
+                    if output.amount != pox_fee {
+                        warn!("Invalid commit tx: different output amounts for different PoX reward addresses");
+                        return Err(op_error::ParseError);
+                    }
+                } else {
+                    pox_fee.replace(output.amount);
+                }
+                commit_outs.push(output.address);
+            }
+        }
+
+        // EITHER there was an amount burned _or_ there were OUTPUTS_PER_COMMIT pox outputs
+        if burn_fee.is_none() && commit_outs.len() != OUTPUTS_PER_COMMIT {
```

Comment:
I thought we were doing both burns and PoX, as part of SIP-007's inevitable transition to PoB?

Reply:
If not, please open an issue so we can later add the PoX --> PoB transition logic in a separate PR.

Reply:
Yep, will open an issue for PoX --> PoB transition/scaling.